frontend/src/hooks/useFetchQuestionByDifficulty.js

```diff
@@ -0,0 +1,32 @@
+import axios from '../api/axios';
+import {useState, useEffect} from 'react'
+import { URL_RETRIEVE_QUESTION_DIFFICULTY } from '../constants';
+
+const useFetchQuestionByDifficulty = (inputDifficulty) => {
+    const [loading, setLoading] = useState(true);
+    const [data, setData] = useState([])
+
+    useEffect((inputDifficulty) => {
+        const fetchData = async () =>{
+          setLoading(true);
+          try {
+            const {data: response} = await axios.post(URL_RETRIEVE_QUESTION_DIFFICULTY,  {
+                difficulty: inputDifficulty || 'medium'
+              })
+            setData(response);
+          } catch (error) {
+            console.error(error);
+          }
+          setLoading(false);
```

Comment:
Should `setLoading(false)` be put inside the `try` block only? I think in case of an error, we shouldn't mark loading as false.